src/ops/conversion.jl

```diff
@@ -0,0 +1,44 @@
+"""
+    convert_value(T, x::Node[; upcast=false]) -> Node
+
+Convert the node `x` to value type `T`, possibly generating a new value.
+
+If and only if `upcast` is `true`, we will always upcast the result of the conversion to
+`T` — the value type of the resulting node will always be `T` in this case.
+
+!!! note
+    By default, this has similar semantics to `Base.convert`, which means that the
+    [`value_type`](@ref) of the returned node might be a subtype of `T`.
+
+    A concrete example:
+    ```julia
```

Comment:
Did you consider making this and the snippet below doctests to ensure that this docstring doesn't go stale?

Reply:
Do you have a reference of how doctests are applied?

I kind of think doctests are a bit evil (since when they exist they have a nasty habit of getting overused / abused instead of "proper" unit tests), but I can look into it. Not in this PR, but we can issueise this if it seems worthwhile.

Reply:
Actually that was pretty painless, and I see that the documentation build is already set to run them. So done :)